bin/oneoff/backfill_data/backfill_script_ids_in_teacher_feedbacks.rb

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+# Backfill existing TeacherFeedbacks to set script_id based on script_level_id.
+
+require_relative '../../../dashboard/config/environment'
+
+def feedbacks_without_script_id
+  TeacherFeedback.where(script_id: nil)
+end
+
+def puts_count
+  puts "There are #{feedbacks_without_script_id.count} feedbacks without a script_id"
+end
+
+def update_script_ids
+  puts "starting backfilling script_ids"
+  feedbacks_without_script_id.find_in_batches(batch_size: 1000) do |batch|
+    print '.'
+    $stdout.flush
+    batch.each do |teacher_feedback|
+      # script_level_id is a required field in teacher_feedbacks, and
+      # script_id is a required field of script_level. We might hit an error
+      # if a script_level_id no longer points to a valid script_level, in which
+      # case we want the script to stop so that we can investigate.
+      script_id = teacher_feedback.script_level.script_id
+      teacher_feedback.update!(script_id: script_id)
```

Comment:
Should we Rescue any Errors raised inside this inner loop, log the TeacherFeedback that couldn't be updated and move on to the next without terminating the entire script?

Reply:
Also, it could be handy to protect the actually data update logic in this line with a "Dry Run" flag, so we can test everything in the script except for the actual update in production before running it for realz. Recent example.